branch-names/action.yml

```diff
@@ -0,0 +1,52 @@
+name: Branch Names
+description: Sanitise Branch names for use
+author: Octopus Deploy
+runs:
+  using: node16
+  main: index.js
+inputs:
+  gitHubRef:
+    description: "This is the input for the github.ref. NOTE: this value should not need to be provided, unless you need to override the default."
```

Comment:
Are there reasons anyone would ever want to override the default? I wonder if we really need to support inputs, or can we make it simple and read the `github.X` directly somehow?

Reply:
I don't know how else to do this. Looking at the [GitHub docs](https://docs.github.com/en/github-ae@latest/actions/security-guides/security-hardening-for-github-actions#using-an-action-instead-of-an-inline-script-recommended), it recommends just passing the Github details in as an input. Do you know of another way?

Reply:
Oh I had assumed we could access the context directly via some function in `@actions/core` but you are right it doesn't look like it. The way you've done it seems fine for this then, the defaults make sense so likely no one will need to override them anyway.

Reply:
I didn't think to look in @actions/core, so I will take a look and use it if it exists otherwise, leave it as is.

Reply:
I took a quick look and nothing stood out, but yeah see what you can find

Reply:
After looking around, it looks like the core package does not contain what we nee, some of the information is available in [this package](https://github.com/actions/toolkit/blob/main/packages/github) but not everything.

Reply:
Hmmm yeah, maybe we should just stick to using the inputs as you've done already

Reply:
I did just check and it looks like we could actually get the information from environment variables. Not sure if GitHub changes those variables very often though. I might just stick to this implementation.